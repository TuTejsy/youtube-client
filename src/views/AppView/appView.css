body {
  background: #2e2e2e;
}

a {
  text-decoration: none;
}

.container {
  max-width: 1440px;
  margin: 0 auto;
}

.logo {
  position: absolute;
  top: 0;
  left: -210px;
	height: 40px;
	text-align: center;
	font-size: 40px;
  font-family: Impact, "Arial Black";
	line-height: 40px;
}

.You {
	color: #fff;
}

.Tube{
	padding: 0.05em .15em;
	border: .05em solid #e23232;
	border-radius: 17.5%;
	background-color:#e23232;
	color: #fff;
}

.search-container {
  position: relative;
  width: 700px;
  height: 40px;
  margin: 20px auto;
  margin-bottom: 100px;
}

.search-container i {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  line-height: 40px;
  font-size: 20px;
  text-align: center;
}
.search-container input {
  display: block;
  width: 650px;
  height: 38px;
  padding-left: 50px;
  font-family: 'Roboto', sans-serif;
  font-size: 20px;
  line-height: 30px;
  color: #1C1C1C;
  border: none;
  border-radius: 5px;
}

.slider {
  width: 1338px;
  height: 450px;
  margin: 0 auto;
  overflow: hidden;
}

.cards-container {
  display: block;
  width: 90000px;
  height: 450px;
  transition: margin-left 0.3s ease-out;
}

.card {
  display: block;
  float: left;
  width: 300px;
  height: 450px;
  margin-right: 46px;
  margin-left: 0;
  margin-top: 0;
  color: hsla(0, 0%, 11%, 0.8);
  background: #fff;
  border-radius: 5px;
}

.card h4 {
  width: 260px;
  margin: 0 auto;
  margin-bottom: 10px;
  text-align: center;
  font-weight: 700;
  font-size: 18px;
}

.card h4 i {
  float: left;
  font-size: 18px;
}

.card p {
  display: block;
  width: 280px;
  margin-top: 20px;
  margin-left: 10px;
  font-size: 16px;
  word-break: break-all;
  word-wrap: break-word;
}

.card .clip-img {
  display: flex;
  align-items: center;
  height: 180px;
  margin-top: 10px;
  margin-bottom: 20px;
  transition: 0.1s ease-in;
}

.card .clip-img h3 {
  padding: 10px;
  width: 280px;
  text-align: center;
  font-family: 'Roboto', sans-serif;
  font-weight: 700;
  font-size: 16px;
  color: #fff;
  background: rgba(166, 0, 0, 0.8);
}

.card .clip-img:hover {
  border-radius: 5px;
  transform: translateY(-10px);
  transition: 0.2s ease-out;
}

.buttons-container {
  display: flex;
  justify-content: space-around;
  width: 200px;
  margin: 50px auto;
}

.buttons-container button {
  margin-right: 20px;
  padding: 0;
  height: 25px;
  width: 25px;
  font-family: 'Roboto', sans-serif;
  font-weight: 700;
  font-size: 15px;
  color: #ff0000;
  border-radius: 50%;
  background: #fff;
  transition: 0.2s ease-in;
}

.buttons-container .active {
  color: #fff;
  border: 1px solid #fff;
  background: #ff0000;
}

.buttons-container .tooltip {
  position: absolute;
  z-index: 20;
  display: none;
  padding: 5px 4px;
  height: 15px;
  color: #1C1C1C;
  font-family: 'Roboto', sans-serif;
  font-weight: 700;
  font-size: 16px;
  line-height: 15px;
  background-color: #fff;
  opacity: 0.8;
  border-radius: 4px;
}


@media only screen and (max-width: 1338px) {
  .slider {
    width: 992px;
  }

  .search-container {
    width: 600px;
  }

  .search-container {
    width: 550px;
  }

  .logo {
    left: -185px;
  }
}

@media only screen and (max-width: 992px) {
  .slider {
    width: 646px;
  }

  .search-container {
    width: 80%;
  }

  .search-container input {
    width: calc(100% - 50px);
  }

  .logo {
    display: none;
  }
}

@media only screen and (max-width: 646px) {
  .slider {
    width: 300px;
  }

  .search-container {
    margin-bottom: 50px;
  }
}
