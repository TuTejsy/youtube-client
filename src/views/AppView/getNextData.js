import AppModel from '../../models/AppModel';
import Request from '../../controllers/createRequest';

export default function getNextData(token) {
  const search = document.querySelector('.search-container input');

  const request = new Request(search.value, token);

  const state = {
    url: request.requestUrl,
  };

  const model = new AppModel(state);
  const data = model.getClips();
  return data;
}
