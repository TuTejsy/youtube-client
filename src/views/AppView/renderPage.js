import Request from '../../controllers/createRequest';
import App from '../../controllers/App';

export default function renderPage() {
  document.head.innerHTML = `${document.head.innerHTML}
      <meta name="viewport" content="width=device-width, initial-scale=1">
      <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
      integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
      crossorigin="anonymous"></link>
      <link href="https://fonts.googleapis.com/css?family=Roboto:400,700&amp;subset=cyrillic"
      rel="stylesheet">`;

  const container = document.createElement('DIV');
  container.className = 'container';

  const search = document.createElement('INPUT');
  search.setAttribute('type', 'text');

  const searchContainer = document.createElement('DIV');
  searchContainer.className = 'search-container';
  searchContainer.innerHTML = `
    <div class="logo">
      <span class="You">You</span>
      <span class="Tube">Tube</span>
    </div>
  <i class="fas fa-search"></i>`;

  const slider = document.createElement('DIV');
  slider.className = 'slider';
  slider.innerHTML = '<div class="cards-container"></div>';

  const buttonsContainer = document.createElement('DIV');
  buttonsContainer.className = 'buttons-container';

  let data;

  function enterPress(e) {
    if (e.keyCode === 13) {
      data = search.value;
      const request = new Request(data);

      const app = new App(request.requestUrl);
      app.start();
    }
  }
  search.addEventListener('keypress', enterPress);

  searchContainer.appendChild(search);
  container.appendChild(searchContainer);
  container.appendChild(slider);
  container.appendChild(buttonsContainer);
  document.body.appendChild(container);
}
