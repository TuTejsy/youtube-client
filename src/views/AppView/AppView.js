import getNextData from './getNextData';

export default class AppView {
  constructor(clip) {
    this.titles = clip.titles;
    this.images = clip.images;
    this.descriptions = clip.descriptions;
    this.authors = clip.authors;
    this.uploadDate = clip.uploadDate;
    this.url = clip.url;
    this.viewCount = clip.viewCount;
    this.nextToken = clip.nextToken;
  }

  async changeThisData(d) {
    const data = await d;
    this.titles = data.titles;
    this.images = data.images;
    this.descriptions = data.descriptions;
    this.authors = data.authors;
    this.uploadDate = data.uploadDate;
    this.url = data.url;
    this.viewCount = data.viewCount;
    this.nextToken = data.nextToken;
  }

  async addCards(count) {
    const cardsContainer = document.querySelector('.cards-container');
    let cards = '';

    for (let i = 0; i < count; i += 1) {
      const uploadDate = this.uploadDate[i].split('').slice(0, 10).join('');

      const card = `
        <div class="card">
          <a class="clip-img"  href="${this.url[i]}" target='_blank' style="background: url('${this.images[i]}') no-repeat center">
            <h3>${this.titles[i]}</h3>
          </a>
          <h4><i class="fas fa-user"></i> ${this.authors[i]}</h4>
          <h4><i class="far fa-calendar-alt"></i> ${uploadDate}</h4>
          <h4><i class="fas fa-eye"></i> ${this.viewCount[i]}</h4>
          <p>${this.descriptions[i]}</p>
        </div>
        `;

      cards += card;
    }

    cardsContainer.innerHTML += cards;
  }

  renderCards(count) {
    const cardsContainer = document.querySelector('.cards-container');
    cardsContainer.innerHTML = '';

    this.addCards(count);

    cardsContainer.style.marginLeft = '0';

    const width = 346;
    let quantity = 4;
    let position = 0;
    let currentPosition = 0;
    let page = 1;

    const buttonsContainer = document.querySelector('.buttons-container');

    const toolTip = document.createElement('DIV');
    toolTip.classList.add('tooltip');

    const firstButton = document.createElement('BUTTON');
    firstButton.innerText = page;
    const secondButton = document.createElement('BUTTON');
    const thirdButton = document.createElement('BUTTON');
    const afterButton = document.createElement('BUTTON');
    afterButton.innerText = page + 1;

    let activeButton = firstButton;
    activeButton.classList.add('active');

    let pageCount = 1;

    let timer;
    function toolTipEvent(e) {
      const button = e.target;

      if (e.clientX) {
        timer = setTimeout(() => {
          toolTip.style.left = `${e.clientX - 10}px`;
          toolTip.style.top = `${e.clientY - 50}px`;

          toolTip.innerText = button.innerText;
          toolTip.style.display = 'block';
        }, 1000);
      } else if (e.targetTouches) {
        timer = setTimeout(() => {
          toolTip.style.left = `${e.targetTouches[0].clientX - 10}px`;
          toolTip.style.top = `${e.targetTouches[0].clientY - 50}px`;

          toolTip.innerText = button.innerText;
          toolTip.style.display = 'block';
        }, 1000);
      }
    }

    function cancelToolTip() {
      clearTimeout(timer);
      toolTip.style.display = 'none';
    }

    firstButton.addEventListener('mousedown', toolTipEvent);
    secondButton.addEventListener('mousedown', toolTipEvent);
    thirdButton.addEventListener('mousedown', toolTipEvent);
    afterButton.addEventListener('mousedown', toolTipEvent);

    firstButton.addEventListener('touchstart', toolTipEvent);
    secondButton.addEventListener('touchstart', toolTipEvent);
    thirdButton.addEventListener('touchstart', toolTipEvent);
    afterButton.addEventListener('touchstart', toolTipEvent);

    document.addEventListener('mouseup', cancelToolTip);
    document.addEventListener('touchend', cancelToolTip);

    function resize() {
      if (document.documentElement.clientWidth < width * 2) {
        quantity = 1;
      } else if (document.documentElement.clientWidth < width * 3) {
        quantity = 2;
      } else if (document.documentElement.clientWidth < width * 4) {
        quantity = 3;
      } else {
        quantity = 4;
      }

      const k = Math.floor(position / (-width * quantity));

      if (k >= 3) {
        currentPosition = position + width * quantity * 3;

        page = k - 2;
        firstButton.innerText = page;
        secondButton.innerText = page + 1;
        thirdButton.innerText = page + 2;
        afterButton.innerText = page + 3;

        activeButton.classList.remove('active');
        activeButton = afterButton;
        activeButton.classList.add('active');
      } else if (k === 2) {
        currentPosition = 0;

        page = k - 1;
        firstButton.innerText = page;
        secondButton.innerText = page + 1;
        thirdButton.innerText = page + 2;
        afterButton.innerText = page + 3;

        activeButton.classList.remove('active');
        activeButton = thirdButton;
        activeButton.classList.add('active');
      } else if (k === 1) {
        currentPosition = 0;

        page = k;
        firstButton.innerText = page;
        secondButton.innerText = page + 1;
        thirdButton.innerText = page + 2;
        afterButton.innerText = page + 3;

        activeButton.classList.remove('active');
        activeButton = secondButton;
        activeButton.classList.add('active');
      }
    }

    resize();

    window.addEventListener('resize', resize);

    firstButton.onclick = () => {
      if (currentPosition !== 0 && position === currentPosition) {
        currentPosition = Math.min(currentPosition + width * quantity, 0);

        page -= 1;
        firstButton.innerText = page;
        secondButton.innerText = page + 1;
        thirdButton.innerText = page + 2;
        afterButton.innerText = page + 3;
      }

      position = currentPosition;
      cardsContainer.style.marginLeft = `${position}px`;

      activeButton.classList.remove('active');
      activeButton = firstButton;
      activeButton.classList.add('active');
    };

    secondButton.onclick = () => {
      position = currentPosition - width * quantity;
      cardsContainer.style.marginLeft = `${position}px`;

      activeButton.classList.remove('active');
      activeButton = secondButton;
      activeButton.classList.add('active');
    };

    thirdButton.onclick = () => {
      position = currentPosition - width * 2 * quantity;
      cardsContainer.style.marginLeft = `${position}px`;


      activeButton.classList.remove('active');
      activeButton = thirdButton;
      activeButton.classList.add('active');
    };

    afterButton.onclick = () => {
      const buttons = buttonsContainer.getElementsByTagName('BUTTON');
      if ((position / -width) + (quantity * 2) > pageCount * count) {
        pageCount += 1;
        const data = getNextData(this.nextToken);

        this.changeThisData(data).then(() => {
          this.addCards(count);
        });
      }

      if (position + width * quantity * 2 < currentPosition) {
        currentPosition -= width * quantity;
        position = currentPosition - width * 3 * quantity;

        page += 1;
        firstButton.innerText = page;
        secondButton.innerText = page + 1;
        thirdButton.innerText = page + 2;
        afterButton.innerText = page + 3;

        activeButton.classList.remove('active');
        activeButton = afterButton;
        activeButton.classList.add('active');
      } else if (buttons.length === 3) {
        thirdButton.innerText = page + 2;
        afterButton.innerText = page + 3;

        buttonsContainer.insertBefore(thirdButton, afterButton);
        position = currentPosition - width * 2 * quantity;

        activeButton.classList.remove('active');
        activeButton = thirdButton;
        activeButton.classList.add('active');
      } else if (buttons.length === 2) {
        secondButton.innerText = page + 1;
        afterButton.innerText = page + 2;

        buttonsContainer.insertBefore(secondButton, afterButton);
        position = currentPosition - width * quantity;

        activeButton.classList.remove('active');
        activeButton = secondButton;
        activeButton.classList.add('active');
      } else if (!afterButton.classList.contains('active')) {
        position = currentPosition - width * 3 * quantity;
        activeButton.classList.remove('active');
        activeButton = afterButton;
        activeButton.classList.add('active');
      }

      cardsContainer.style.marginLeft = `${position}px`;
    };

    const currentThis = this;

    function cardsContainerClick(e) {
      if (e.target.classList.contains('card-img')) return;
      let firstPos;
      let secondPos;
      let pos = position;

      if (e.clientX) {
        firstPos = e.clientX;
      } else if (e.targetTouches) {
        firstPos = e.targetTouches[0].clientX;
      }

      function move(ev) {
        if (ev.clientX) {
          secondPos = ev.clientX;
        } else if (ev.targetTouches) {
          secondPos = ev.targetTouches[0].clientX;
        }
        pos += (secondPos - firstPos) * 1.5;
        cardsContainer.style.marginLeft = `${pos}px`;
      }

      function cardsContainerMove(ev) {
        move(ev);
        firstPos = secondPos;
      }

      cardsContainer.addEventListener('mousemove', cardsContainerMove);
      cardsContainer.addEventListener('touchmove', cardsContainerMove);

      buttonsContainer.insertBefore(secondButton, afterButton);
      buttonsContainer.insertBefore(thirdButton, afterButton);
      firstButton.innerText = page;
      secondButton.innerText = page + 1;
      thirdButton.innerText = page + 2;
      afterButton.innerText = page + 3;

      function cancelMove() {
        cardsContainer.removeEventListener('mousemove', cardsContainerMove);
        cardsContainer.removeEventListener('touchmove', cardsContainerMove);

        if ((position / -width) + quantity + 2 > pageCount * count) {
          pageCount += 1;
          const data = getNextData(this.nextToken);

          currentThis.changeThisData(data).then(() => {
            currentThis.addCards(count);
          });
        }

        position = Math.round(pos / width) * width;

        const k = Math.floor(position / (-width * quantity));

        if (k >= 3 || afterButton.classList.contains(!'active')) {
          currentPosition = position + width * quantity * 3;

          page = k - 2;
          firstButton.innerText = page;
          secondButton.innerText = page + 1;
          thirdButton.innerText = page + 2;
          afterButton.innerText = page + 3;

          activeButton.classList.remove('active');
          activeButton = afterButton;
          activeButton.classList.add('active');
        } else if (k === 2) {
          currentPosition = 0;

          page = k - 1;
          firstButton.innerText = page;
          secondButton.innerText = page + 1;
          thirdButton.innerText = page + 2;
          afterButton.innerText = page + 3;

          activeButton.classList.remove('active');
          activeButton = thirdButton;
          activeButton.classList.add('active');
        } else if (k === 1) {
          currentPosition = 0;

          page = k;
          firstButton.innerText = page;
          secondButton.innerText = page + 1;
          thirdButton.innerText = page + 2;
          afterButton.innerText = page + 3;

          activeButton.classList.remove('active');
          activeButton = secondButton;
          activeButton.classList.add('active');
        } else {
          currentPosition = 0;
          page = 1;

          firstButton.innerText = page;
          secondButton.innerText = page + 1;
          thirdButton.innerText = page + 2;
          afterButton.innerText = page + 3;
          activeButton.classList.remove('active');
          activeButton = firstButton;
          activeButton.classList.add('active');
        }
        if (position > 0) {
          position = 0;
        }

        cardsContainer.style.marginLeft = `${position}px`;

        document.removeEventListener('mouseup', cancelMove);
        document.removeEventListener('touchend', cancelMove);
      }

      document.addEventListener('mouseup', cancelMove);
      document.addEventListener('touchend', cancelMove);
    }

    cardsContainer.addEventListener('mousedown', cardsContainerClick);
    cardsContainer.addEventListener('touchstart', cardsContainerClick);

    buttonsContainer.innerHTML = '';
    buttonsContainer.appendChild(firstButton);
    buttonsContainer.appendChild(afterButton);
    buttonsContainer.appendChild(toolTip);
  }
}
