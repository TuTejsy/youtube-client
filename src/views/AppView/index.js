import AppView from './AppView';

import './reset.css';
import './appView.css';

export default AppView;
