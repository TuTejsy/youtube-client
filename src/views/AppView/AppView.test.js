import AppView from './AppView';

describe('AppView.prototype.changeThisData', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.changeThisData).toBeInstanceOf(Function);
  });

  it('Should change clip properties', () => {
    const data = {
      titles: ['We Did a BIG MISTAKE !!', 'this Happened with my HAYABUSA in NEPAL', 'JS ft. R Kelly - Love Angel'],
      images: ['https://i.ytimg.com/vi/qBDU4vcILhg/mqdefault.jpg', 'https://i.ytimg.com/vi/SJKMpODfLfU/mqdefault.jpg', 'https://i.ytimg.com/vi/7QkCrmq3LYc/mqdefault.jpg'],
      descriptions: [
        'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH --------------------------------------------------------------- ***Buy my CALENDAR and other ...',
        'EP11 #bhutanonbusa #indiatobhutan Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH Road Gods : https://amzn.to/2Xy0786 Rynox Gears ...',
        'Love Angel music video taken from the Ice Cream album.',
      ],
      authors: ['JS Films', 'JS Films', 'askmeifimalemon'],
      uploadDate: ['2019-05-07T03:30:04.000Z', '2019-05-03T03:30:01.000Z', '2007-02-06T03:50:56.000Z'],
      url: ['https://www.youtube.com/watch?v=qBDU4vcILhg', 'https://www.youtube.com/watch?v=SJKMpODfLfU', 'https://www.youtube.com/watch?v=7QkCrmq3LYc'],
      viewCount: ['2148846', '5662221', '268158'],
      nextToken: 'CA8QAA',
    };

    const view = new AppView(data);

    const newData = {
      titles: ['We Did', 'HAYABUSA in NEPAL', 'Love Angel'],
      images: ['https://i.ytimg.com/vi/qBDU4sfdgewLhg/mqdefault.jpg', 'https://i.ytimg.com/vi/sdfghMpODfLfU/mqdefault.jpg', 'https://i.ytimg.com/vi/7QkCdsfghq3LYc/mqdefault.jpg'],
      descriptions: [
        'Interesting video',
        'Look at ME!',
        'Love Angel music',
      ],
      authors: ['Films', 'JS', 'TUTEJSY'],
      uploadDate: ['2000-07-07T03:30:04.000Z', '2009-09-03T03:30:01.000Z', '2017-02-06T03:50:56.000Z'],
      url: ['https://www.youtube.com/watch?v=qBSDF4vcILhg', 'https://www.youtube.com/watch?v=SJKMpSDFfLfU', 'https://www.youtube.com/watch?v=7QkCrASDFc'],
      viewCount: ['21446', '5662223421', '2682342358'],
      nextToken: 'CA8QBB',
    };

    view.changeThisData(newData).then(() => {
      expect(view.titles).toEqual(newData.titles);
      expect(view.images).toEqual(newData.images);
      expect(view.descriptions).toEqual(newData.descriptions);
      expect(view.authors).toEqual(newData.authors);
      expect(view.uploadDate).toEqual(newData.uploadDate);
      expect(view.url).toEqual(newData.url);
      expect(view.viewCount).toEqual(newData.viewCount);
      expect(view.nextToken).toBe(newData.nextToken);
    });
  });
});

describe('AppView.prototype.constructor', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.constructor).toBeInstanceOf(Function);
  });

  it('Should set clip properties', () => {
    const data = {
      titles: ['We Did a BIG MISTAKE !!', 'this Happened with my HAYABUSA in NEPAL', 'JS ft. R Kelly - Love Angel'],
      images: ['https://i.ytimg.com/vi/qBDU4vcILhg/mqdefault.jpg', 'https://i.ytimg.com/vi/SJKMpODfLfU/mqdefault.jpg', 'https://i.ytimg.com/vi/7QkCrmq3LYc/mqdefault.jpg'],
      descriptions: [
        'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH --------------------------------------------------------------- ***Buy my CALENDAR and other ...',
        'EP11 #bhutanonbusa #indiatobhutan Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH Road Gods : https://amzn.to/2Xy0786 Rynox Gears ...',
        'Love Angel music video taken from the Ice Cream album.',
      ],
      authors: ['JS Films', 'JS Films', 'askmeifimalemon'],
      uploadDate: ['2019-05-07T03:30:04.000Z', '2019-05-03T03:30:01.000Z', '2007-02-06T03:50:56.000Z'],
      url: ['https://www.youtube.com/watch?v=qBDU4vcILhg', 'https://www.youtube.com/watch?v=SJKMpODfLfU', 'https://www.youtube.com/watch?v=7QkCrmq3LYc'],
      viewCount: ['2148846', '5662221', '268158'],
      nextToken: 'CA8QAA',
    };

    const view = new AppView(data);

    expect(view.titles).toEqual(data.titles);
    expect(view.images).toEqual(data.images);
    expect(view.descriptions).toEqual(data.descriptions);
    expect(view.authors).toEqual(data.authors);
    expect(view.uploadDate).toEqual(data.uploadDate);
    expect(view.url).toEqual(data.url);
    expect(view.viewCount).toEqual(data.viewCount);
    expect(view.nextToken).toBe(data.nextToken);
  });
});

describe('AppView.prototype.renderCards', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.renderCards).toBeInstanceOf(Function);
  });

  it('Should render cards', () => {
    const data = {
      titles: ['We Did a BIG MISTAKE !!', 'this Happened with my HAYABUSA in NEPAL', 'JS ft. R Kelly - Love Angel'],
      images: ['https://i.ytimg.com/vi/qBDU4vcILhg/mqdefault.jpg', 'https://i.ytimg.com/vi/SJKMpODfLfU/mqdefault.jpg', 'https://i.ytimg.com/vi/7QkCrmq3LYc/mqdefault.jpg'],
      descriptions: [
        'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH --------------------------------------------------------------- ***Buy my CALENDAR and other ...',
        'EP11 #bhutanonbusa #indiatobhutan Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH Road Gods : https://amzn.to/2Xy0786 Rynox Gears ...',
        'Love Angel music video taken from the Ice Cream album.',
      ],
      authors: ['JS Films', 'JS Films', 'askmeifimalemon'],
      uploadDate: ['2019-05-07T03:30:04.000Z', '2019-05-03T03:30:01.000Z', '2007-02-06T03:50:56.000Z'],
      url: ['https://www.youtube.com/watch?v=qBDU4vcILhg', 'https://www.youtube.com/watch?v=SJKMpODfLfU', 'https://www.youtube.com/watch?v=7QkCrmq3LYc'],
      viewCount: ['2148846', '5662221', '268158'],
      nextToken: 'CA8QAA',
    };

    const container = document.createElement('DIV');
    container.className = 'container';

    const search = document.createElement('INPUT');
    search.setAttribute('type', 'text');

    const searchContainer = document.createElement('DIV');
    searchContainer.className = 'search-container';
    searchContainer.innerHTML = `
    <div class="logo">
      <span class="You">You</span>
      <span class="Tube">Tube</span>
    </div>
  <i class="fas fa-search"></i>`;

    const slider = document.createElement('DIV');
    slider.className = 'slider';
    slider.innerHTML = '<div class="cards-container"></div>';

    const buttonsContainer = document.createElement('DIV');
    buttonsContainer.className = 'buttons-container';

    searchContainer.appendChild(search);
    container.appendChild(searchContainer);
    container.appendChild(slider);
    container.appendChild(buttonsContainer);
    document.body.appendChild(container);

    const view = new AppView(data);
    view.renderCards(5);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});

describe('AppView.prototype.addCards', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.addCards).toBeInstanceOf(Function);
  });

  it('Should add cards', () => {
    const data = {
      titles: ['We Did a BIG MISTAKE !!', 'this Happened with my HAYABUSA in NEPAL', 'JS ft. R Kelly - Love Angel'],
      images: ['https://i.ytimg.com/vi/qBDU4vcILhg/mqdefault.jpg', 'https://i.ytimg.com/vi/SJKMpODfLfU/mqdefault.jpg', 'https://i.ytimg.com/vi/7QkCrmq3LYc/mqdefault.jpg'],
      descriptions: [
        'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH --------------------------------------------------------------- ***Buy my CALENDAR and other ...',
        'EP11 #bhutanonbusa #indiatobhutan Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH Road Gods : https://amzn.to/2Xy0786 Rynox Gears ...',
        'Love Angel music video taken from the Ice Cream album.',
      ],
      authors: ['JS Films', 'JS Films', 'askmeifimalemon'],
      uploadDate: ['2019-05-07T03:30:04.000Z', '2019-05-03T03:30:01.000Z', '2007-02-06T03:50:56.000Z'],
      url: ['https://www.youtube.com/watch?v=qBDU4vcILhg', 'https://www.youtube.com/watch?v=SJKMpODfLfU', 'https://www.youtube.com/watch?v=7QkCrmq3LYc'],
      viewCount: ['2148846', '5662221', '268158'],
      nextToken: 'CA8QAA',
    };

    const container = document.createElement('DIV');
    container.className = 'container';

    const slider = document.createElement('DIV');
    slider.className = 'slider';
    slider.innerHTML = '<div class="cards-container"></div>';

    const buttonsContainer = document.createElement('DIV');
    buttonsContainer.className = 'buttons-container';

    container.appendChild(slider);
    document.body.appendChild(container);

    const view = new AppView(data);
    view.addCards(5);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
