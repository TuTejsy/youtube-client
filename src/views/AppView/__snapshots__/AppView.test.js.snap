// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.addCards Should add cards 1`] = `
"<div class=\\"container\\"><div class=\\"search-container\\">
    <div class=\\"logo\\">
      <span class=\\"You\\">You</span>
      <span class=\\"Tube\\">Tube</span>
    </div>
  <i class=\\"fas fa-search\\"></i><input type=\\"text\\"></div><div class=\\"slider\\"><div class=\\"cards-container\\" style=\\"margin-left: 0px;\\"></div></div><div class=\\"buttons-container\\"><button class=\\"active\\"></button><button></button></div></div><div class=\\"container\\"><div class=\\"slider\\"><div class=\\"cards-container\\"></div></div></div>"
`;

exports[`AppView.prototype.renderCards Should render cards 1`] = `
"<div class=\\"container\\"><div class=\\"search-container\\">
    <div class=\\"logo\\">
      <span class=\\"You\\">You</span>
      <span class=\\"Tube\\">Tube</span>
    </div>
  <i class=\\"fas fa-search\\"></i><input type=\\"text\\"></div><div class=\\"slider\\"><div class=\\"cards-container\\" style=\\"margin-left: 0px;\\"></div></div><div class=\\"buttons-container\\"><button class=\\"active\\"></button><button></button></div></div>"
`;
